gears/src/types/tx/mod.rs

```diff
@@ -26,6 +29,53 @@ pub trait TxMessage:
     fn type_url(&self) -> &'static str;
 }
 
+/// Utility type that guarantees correctness of transaction messages set
+pub struct Messages<T: TxMessage> {
+    messages: Vec<T>,
```

Comment:
Not sure, that you want, but there `Vec1` type

Reply:
We suggested to save this struct.